processor/src/main/java/org/mapstruct/ap/internal/model/source/SelectionParameters.java

```diff
@@ -67,9 +71,8 @@ public TypeMirror getResultType() {
     @Override
     public int hashCode() {
         int hash = 3;
-        hash = 97 * hash + (this.qualifiers != null ? this.qualifiers.hashCode() : 0);
         hash = 97 * hash + (this.qualifyingNames != null ? this.qualifyingNames.hashCode() : 0);
-        hash = 97 * hash + (this.resultType != null ? this.resultType.hashCode() : 0);
+        hash = 97 * hash + (this.resultType != null ? this.resultType.toString().hashCode() : 0);
```

Comment:
Which name do you mean? The `hashCode` is also not deterministic for `TypeMirror` and the `resultType` is a `TypeMirror`